src/main/java/emu/grasscutter/game/quest/QuestManager.java

```diff
@@ -227,29 +230,95 @@ public GameQuest addQuest(int questId) {
         return quest;
     }
 
-    public void startMainQuest(int mainQuestId) {
-        var mainQuestData = GameData.getMainQuestDataMap().get(mainQuestId);
+    public void startMainQuest(boolean notify, int mainQuestId, boolean useInput, String paramStr, int... params) {
+        MainQuestData mainQuestData = GameData.getMainQuestDataMap().get(mainQuestId);
+
+        if (mainQuestData == null) return; // check for bin output
+
+        // Main quest
+        GameMainQuest mainQuest = getMainQuestById(mainQuestId);
+
+        // Create main quest if it doesnt exist
+        if (mainQuest == null) {
+            mainQuest = addMainQuest(mainQuestId);
+        } else {
+            return;
+        }
+
+        int[] subQuestBegun = new int[(int) mainQuest.getChildQuests().values().stream().count()];
+
+        for (GameQuest subQuest : mainQuest.getChildQuests().values().stream().toList()) {
+            val acceptCond = subQuest.getQuestData().getAcceptCond();
+            boolean shouldAccept = false;
+            if (acceptCond.size() == 0) {
+                // generally means QUEST_COND_NONE
+                shouldAccept = true;
+            } else {
+                int[] accept = new int[acceptCond.size()];
+
+                for (int i = 0; i < acceptCond.size(); i++) {
+                    val condition = acceptCond.get(i);
+                    int[] condParam = condition.getParam();
+                    String condParamStr = condition.getParamStr();
+                    if (useInput) { // differentiate usage for queue event
```

Comment:
This kinda looks like a workaround, if this happens with the wrong conditions, this will always result in success, which I'm certain it should not.
The condition checks probably need changes